### DISCUSSION TITLE
Semantic Versioning compliance.

### DISCUSSION THREAD

**Discussion by olivier-spinelli** [emgucv/emgucv#411] [Ideas]
Hello,

The current version pattern is not SemVer compliant (https://semver.org) since it uses a 4 parts identifier (Emgu.CV.runtime.windows 4.5.1.4349).

The global tendancy is to use Semantic Versioning: more and more tools rely on it and it's diffcult (sometimes impossible) to handle 4 parts versions with some of them.

Is there any chance for EgmuCV to be SemVer compliant?

Thanks in advance for any answer.

**Comment by emgucv** [comment]
Thanks for pointing that out.

I am assuming you are taking about versioning for nuget pacakges. Is that correct?

Emgu CV use 4 part versioning for nuget packages similar to Xamarin Forms:
https://www.nuget.org/packages/Xamarin.Forms/

> The global tendancy is to use Semantic Versioning: more and more tools rely on it and it's diffcult (sometimes impossible) to handle 4 parts versions with some of them.

Can you point out some of the example of applications that cannot handle 4 parts versions?

On the other hand, changing our nuget package version from {major}.{minor}.{patch}.{build} to {major}.{minor}.{patch}+{build} should make the nuget package version compatible with Semantic Versioning, am I correct? I wonder why Xamarin Forms, now managed by Microsoft, not use {major}.{minor}.{patch}+{build} version format...
